wallet/src/de/schildbach/wallet/AppDatabase.java

```diff
@@ -36,12 +41,14 @@
 @Database(entities = {
         ExchangeRate.class,
         BlockchainState.class,
+        TransactionMetadata.class,
+        AddressMetadata.class,
         BlockchainIdentityData.class,
         DashPayProfile.class,
         DashPayContactRequest.class,
         UserAlert.class,
         Invitation.class
-}, version = 16)
+}, version = 12)
```

Comment:
This looks strange.  DashPay was on 16, while master was on 11. 

In our next release of DashPay, there will be a complete data wipe on Platfrom, so none of the data will need to be saved, though it may cause the app to crash.

Reply:
This will surely cause an upgraded app to crash at the start, so DashPay users will have to reinstall it. Is that fine?
Maybe we can set the version to 17, it shouldn't be a problem imo.

Reply:
That is a good question.  When we release a new DashPay app, we are not expecting the old data to be usable.  We will tell users to uninstall, reinstall, but some will not.  If it crashes, the user may give up on the app instead of uninstalling and reinstalling.

Although Version 16, won't have the tables from version 11.

Dash Wallet 10->11 Add metadata tables - after the next master merge into DashPay.
DashPay 11->16, add dashpay tables, but leave out metadata tables
DashPay 16->17 - Add metadata tables.

For the purposes of DashPay on testnet, this is probably fine, but when considering an upgrade on mainnet, where we upgrade from DashWallet to DashPay, it seems like there might be room for losing metadata tables.  If we go with v17 now, I should put notes about changing it later?

Reply:
When we upgrade the Wallet to DashPay, we will need to write migrations for adding DashPay tables. It will be a migration from whatever version the wallet has to whatever version DashPay has at that time.
We'll need these migrations anyway if we plan to do any changes in the Wallet database.
I'm sure the upgrade process will be tested thoufully.

Reply:
Ok, then I will return this version to 17.  When we do the upgrade from Dash Wallet to DashPay, then the need for the DashPay Testnet app will probably be reduced significantly and we won't need to support it as much.  If something is broken on the testnet side with older versions, then that is probably fine.